### ISSUE TITLE
Add color module support to Basis

### ISSUE THREAD

**Issue by hosef** [backdrop/backdrop-issues#3028]
## Describe your issue or idea
The default theme for Backdrop does not support the Color module.
We should add support for it to allow more customization of Backdrop out of the box and to have a reference implementation of Color module support available in Basis.

---

PR by @hosef: https://github.com/backdrop/backdrop/pull/2132

**Comment by hosef** [comment]
I have run across a problem setting up Color module support with the fact that Basis has ~55 unique colors, and some things that look like they are the same color are actually different.

**Comment by laryn** [comment]
In that case it would seem to me that standardizing and reducing would be a good idea – @wesruv?

**Comment by docwilmot** [comment]
Started this a while back. Maybe you can take over and create a proper release when youre done.

**Comment by jenlampton** [comment]
@hosef  is it https://github.com/backdrop/backdrop/pull/2132 that needs the review?

**Comment by quicksketch** [comment]
Nice, well https://github.com/backdrop/backdrop/pull/2132 does indeed seem to work. I successfully made a fair atrocious color scheme:

![image](https://user-images.githubusercontent.com/100206/39449848-0697baac-4c7e-11e8-932e-e0b03349de24.png)

![image](https://user-images.githubusercontent.com/100206/39449863-10f4367e-4c7e-11e8-85ae-769eecace435.png)

Some things we'll need to get in place:

- The preview, which I'm guessing is going to be a bit tricky.
- Some default color schemes.
- Perhaps pairing down the color choices a bit? Color module can do variations on the same color I believe. e.g. Take one color and have it with slightly less light and saturation. So we might not need every possible color.

Overall seems to work though! I know @wesruv wanted to add color support but got overwhelmed with the ancient code in Color module.

**Comment by quicksketch** [comment]
BTW, total code quality is top notch. Code-wise this puts in a great place. We just need to complete the implementation. If simplify down a bit if possible, and get @wesruv's opinion those would also be wins to move this forward.

**Comment by hosef** [comment]
I have removed a couple of options, and renamed some of them. I also enabled the color shifting, however it does some weird things sometimes because it only supports having 1 base color and 1 blend target for all colors to use.

Currently, the color shifting code is trying to replicate the color shifting from GD,which was used to handle rounded corners and gradients before CSS supported those things. I personally think it would be better if we removed the GD integration and converted the color shifting to be a simple HSL comparison between 2 base colors.
